shell/renderer/api/atom_api_context_bridge.cc

```diff
@@ -0,0 +1,473 @@
+    const v8::Local<v8::Context>& target_context,
+    context_bridge::RenderFramePersistenceStore* store,
+    int recursion_depth) {
+  mate::Dictionary api(source_context->GetIsolate(), api_object);
+  mate::Dictionary proxy =
+      mate::Dictionary::CreateEmpty(target_context->GetIsolate());
+  store->CacheProxiedObject(api.GetHandle(), proxy.GetHandle());
+  auto maybe_keys = api.GetHandle()->GetOwnPropertyNames(source_context);
+  if (maybe_keys.IsEmpty())
+    return v8::MaybeLocal<v8::Object>(proxy.GetHandle());
+  auto keys = maybe_keys.ToLocalChecked();
+
+  v8::Context::Scope target_context_scope(target_context);
+  uint32_t length = keys->Length();
+  std::string key_str;
+  int key_int;
+  for (uint32_t i = 0; i < length; i++) {
+    v8::Local<v8::Value> key = keys->Get(target_context, i).ToLocalChecked();
+    // Try get the key as a string
+    if (!mate::ConvertFromV8(api.isolate(), key, &key_str)) {
+      // Try get the key as an int
+      if (!mate::ConvertFromV8(api.isolate(), key, &key_int))
+        continue;
+      // Convert the int to a string as they are interoperable as object keys
+      key_str = base::NumberToString(key_int);
+    }
+    v8::Local<v8::Value> value;
+    if (!api.Get(key_str, &value))
+      continue;
+
+    auto passed_value = PassValueToOtherContext(
+        source_context, target_context, value, store, recursion_depth + 1);
+    if (passed_value.IsEmpty())
+      return v8::MaybeLocal<v8::Object>();
+    proxy.Set(key_str, passed_value.ToLocalChecked());
+  }
+
+  return proxy.GetHandle();
+}
+
+#ifdef DCHECK_IS_ON
+mate::Dictionary DebugGC(mate::Dictionary empty) {
+  auto* render_frame = GetRenderFrame(empty.GetHandle());
+  auto* store = GetOrCreateStore(render_frame);
+  mate::Dictionary ret = mate::Dictionary::CreateEmpty(empty.isolate());
+  ret.Set("functionCount", store->functions().size());
+  auto* proxy_map = store->proxy_map();
+  ret.Set("objectCount", proxy_map->size() * 2);
+  int live_from = 0;
+  int live_proxy = 0;
+  for (auto iter = proxy_map->begin(); iter != proxy_map->end(); iter++) {
+    auto* node = iter->second;
+    while (node) {
+      if (!std::get<0>(node->pair).IsEmpty())
+        live_from++;
+      if (!std::get<1>(node->pair).IsEmpty())
+        live_proxy++;
+      node = node->next;
+    }
+  }
+  ret.Set("liveFromValues", live_from);
+  ret.Set("liveProxyValues", live_proxy);
+  return ret;
+}
+#endif
+
+void ExposeAPIInMainWorld(const std::string& key,
+                          v8::Local<v8::Object> api_object,
+                          mate::Arguments* args) {
+  auto* render_frame = GetRenderFrame(api_object);
+  CHECK(render_frame);
+  context_bridge::RenderFramePersistenceStore* store =
+      GetOrCreateStore(render_frame);
+  auto* frame = render_frame->GetWebFrame();
+  CHECK(frame);
+  v8::Local<v8::Context> main_context = frame->MainWorldScriptContext();
+  mate::Dictionary global(main_context->GetIsolate(), main_context->Global());
```

Comment:
`gin_helper::Dictionary`

Reply:
Same comment as above RE `mate` and backporting